### ISSUE TITLE
Crash: Merge Table

### ISSUE THREAD

**Issue by GeoDaTester** [GeoDaCenter/geoda#1491]
From: 
 Details:
os: 3-4-4
vs: 1-12-1-57
/usr/local/geoda/web_plugins/logger.txt
 GdaFrame::ShowOpenDatasourceDlg()
ConnectDatasourceDlg::InitSamplePanel()
Check auto update:
AutoUpdate::CheckUpdate()
AutoUpdate::GetCheckList()
AutoUpdate::ReadUrlContent()
Entering ConnectDatasourceDlg::OnOkClick
ConnectDatasourceDlg::CreateDataSource()
Open Datasource:/media/agrs1mg0701ce02/KINGSTON/GEODA/shape_prv/inec_prv.xls
Open Layer:inec_prv
Entering ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::OnOkClick
Entering Project::Project (new project)
Project::CommonProjectInit()
Entering Project::InitFromOgrLayer
Datasource name:
/media/agrs1mg0701ce02/KINGSTON/GEODA/shape_prv/inec_prv.xls
Project::GetSpatialReference()
Exiting Project::Project
Click GdaFrame::InitWithProject()
Open TableFrame.
In TableFrame::OnActivate
In TableFrame::OnActivate
In TableFrame::OnActivate
In TableFrame::OnActivate
Click GdaFrame::OnCloseProjectEvt
Project::GetProjectFullPath()
In TableFrame::OnClose and actually closing.
Project::CleanupPairsHLState()
Entering Project::~Project
Exiting Project::~Project
Click GdaFrame::OnNewProject
 GdaFrame::ShowOpenDatasourceDlg()
ConnectDatasourceDlg::InitSamplePanel()
Entering ConnectDatasourceDlg::OnOkClick
ConnectDatasourceDlg::CreateDataSource()
Open Datasource:/media/agrs1mg0701ce02/KINGSTON/GEODA/shape_prv/nxprovincias.shp
Open Layer:nxprovincias
Entering ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::SaveRecentDataSource
Exiting ConnectDatasourceDlg::OnOkClick
Entering Project::Project (new project)
Project::CommonProjectInit()
Entering Project::InitFromOgrLayer
Datasource name:
/media/agrs1mg0701ce02/KINGSTON/GEODA/shape_prv/nxprovincias.shp
Project::GetSpatialReference()
Exiting Project::Project
Click GdaFrame::InitWithProject()
Open TableFrame.
Open MapFrame.
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
In TableFrame::OnActivate
In TableFrame::OnActivate
Open MergeTableDlg.
Entering MergeTableDlg::OnOpenClick()
Entering MergeTableDlg::OnOpenClick()
In MergeTableDlg::OnClose()
In TableFrame::OnActivate
In TableFrame::OnActivate
Open MergeTableDlg.
Entering MergeTableDlg::OnOpenClick()
Entering MergeTableDlg::OnOpenClick()

**Comment by jkoschinsky** [comment]
@DarcyLinde 
please test merge table functionality with "unclean" data, e.g. missing values, non-standard formats etc. -- could use data from open data portal. thx

**Comment by Ashitacarl** [comment]
GeoDa 1.12.1.191 (macOS Mojave, 10.14.1). Jan. 16 build.

Tested using datasets with missing values, numeric and character key variables. No crash.

@lixun910 It seems GeoDa cannot merge tables based on non-English key variables. Is this the case?

**Comment by lixun910** [comment]
Thanks, @Ashitacarl !

It can merge tables based on non-english key variables. But one has to select proper encoding first. For example:

1. load table with NAME field in Chinese, and select GB2312 encoding to display chinese characters correctly
<img width="388" alt="screen shot 2019-01-18 at 5 34 15 pm" src="https://user-images.githubusercontent.com/2423887/51419440-c009b380-1b47-11e9-8c6f-ff7d88bc7cae.png">

2. in merge, select proper encoding (e.g. "GB2312") before drag-n-drop your second data. 
<img width="1047" alt="screen shot 2019-01-18 at 5 35 19 pm" src="https://user-images.githubusercontent.com/2423887/51419473-ff380480-1b47-11e9-9865-9f12e05d71e5.png">

Then, you can do merge using the non-english key variables from working table and merging table.

**Comment by Ashitacarl** [comment]
@lixun910  It works. Thanks! 

But when I tried to make changes to the characters, some weird things happened. 

After encoding to Chinese Simplified GB2312, I tried to change “北京市” to “北京” or the other way around. But it seems to result in some unrecognizable character (I assume it is in another encoding system). This only occurs for existing variables. If I create a new variable, I don't see this problem. See below. Is this a problem or am I doing it in a wrong way? 

<img width="334" alt="after" src="https://user-images.githubusercontent.com/45187310/51424221-945eeb80-1b90-11e9-8c00-d4bf58fd84f9.png">
